Add rpkamp-go

rpkamp opened the pull request:
Add a raffler in go that uses goroutines and a lot of randomness.

![Goroutines, Goroutines everywhere](https://i.imgflip.com/yf2u6.jpg)

lucasvanlierop commented:
Cool!, would you mind add some docker config for it and adding it to:
* https://github.com/domcode/rafflers/blob/master/build.sh
* https://github.com/domcode/rafflers/blob/master/raffle.sh

Check one of the existing rafflers for inspiration, or ping me directly.

rpkamp commented:
No problem, I'll add it tonight :)

lucasvanlierop commented:
Thanks!

rpkamp commented:
Updated.

I've also changed `build.sh` so it will automatically detect what to build, and build it - skipping containers that were already build.
I've also changed `raffle.sh` so it will automatically detect from docker what rafflers are available, so there is no more need to hardcode them.

lucasvanlierop commented on build.sh:
```diff
@@ -1,9 +1,13 @@
-docker build -t lucasvanlierop_cobol_raffler lucasvanlierop-cobol
-docker build -t rjkip_elixir_raffler rjkip-elixir
-docker build -t sgoettschkes_haskell_raffler sgoettschkes-haskell
+dockerfiles=$(ls */Dockerfile)
```
The current setup was just to get thing rolling, I was planning to improve it but thanks for fixing it!

lucasvanlierop commented on raffle.sh:
```diff
@@ -11,11 +16,11 @@ NAMES_FILE=`realpath $WORKING_DIR/$1`
 
 # Raffle a raffler ;-)
-declare -a RAFFLER_NAMES=('basbl_zsh' 'lucasvanlierop_cobol' 'rjkip_elixir' 'sgoettschkes_haskell' 'shawnmccool_scala' 'aochagavia_rust')
+declare RAFFLER_NAMES=(`docker images | awk '{print $1}' | grep '_raffler$'`)
```
Nice!

lucasvanlierop commented:
If you can just fix the merge issues then

![http://cdn.meme.am/instances/55018482.jpg](http://cdn.meme.am/instances/55018482.jpg)

rpkamp commented:
Fixed merge conflicts and checked (and fixed) scripts using shellcheck.

![merge](http://s2.quickmeme.com/img/7d/7daec2ce325822b7d22b6d62bca74abbda407ec4cbfa0097d93209d1be77c0a7.jpg)

lucasvanlierop commented:
Jeeej!
